### PR TITLE
Marjorie lv/binary instructions

### PR DESCRIPTION
**Pull request by MarjorieLV**

### CODE REVIEW

File: includes/binaryinstructions.hpp [hunk 1]
```diff
@@ -1,28 +1,678 @@
 #ifndef _BINARYINSTRUCTIONS_
-#define _BINARYINSTRUCTIONS_
+#define _INSTRUCTIONS_
```
Comment (Sgregory42) [inline includes/binaryinstructions.hpp]: should be _BINARYINSTRUCTIONS_

File: includes/binaryinstructions.hpp [hunk 2]
```diff
@@ -1,28 +1,678 @@
...
+    SET (int cycles, int bit, IMemory::REG8BIT reg)
+        :IInstructions(cycles),
+         _bit(bit),
+         _reg8Bit(reg){};
+
+    void doInstruction(IMemory& memory) override {
+        memory.setBitInRegister(_bit, _reg8Bit);
+    }
+
+    int _bit;
```
Comment (Sgregory42) [inline includes/binaryinstructions.hpp:21]: private ?

File: includes/binaryinstructions.hpp [hunk 3]
```diff
@@ -1,28 +1,678 @@
...
+    void doInstruction(IMemory& memory) override {
+        uint16_t adress = memory.get16BitRegister(IMemory::REG16BIT::HL);
+        uint8_t value = memory.readInMemory(adress);
+        std::bitset<8> bitsetValue(value);
+        bitsetValue[_bit] = 1;
+        uint8_t newValue = static_cast<uint8_t>(bitsetValue.to_ulong());
+        memory.writeInMemory(newValue, adress);
+    }
+
+    int _bit;
```
Comment (Sgregory42) [inline includes/binaryinstructions.hpp:42]: private ?

File: includes/binaryinstructions.hpp [hunk 4]
```diff
@@ -1,28 +1,678 @@
...
+    RES (int cycles, int bit, IMemory::REG8BIT reg)
+        :IInstructions(cycles),
+         _bit(bit),
+         _reg8Bit(reg){};
+
+    void doInstruction(IMemory& memory) override {
+        memory.unsetBitInRegister(_bit, _reg8Bit);
+    }
+
+    int _bit;
```
Comment (Sgregory42) [inline includes/binaryinstructions.hpp:58]: private ?

File: src/romloader.cpp [hunk 5]
```diff
@@ -15,7 +15,7 @@ bool RomLoader::load(std::string const & romName)
     if (fd > 0) {
         _data.fill(0);
         uint8_t buff[512];
-        int readCount = 0;
+        int readCount = 0x0100;
```
Comment (Sgregory42) [inline src/romloader.cpp:19]: is this correct ? if so maybe use a variable to let us know what is it. (don't use magic numbers)

File: test/romloader.t.cpp [hunk 6]
```diff
@@ -47,11 +47,14 @@ TEST_F (RomLoaderTest, loadValidFile)
     bool isLoaded = romLoader.load(fileName);
 
     IMemory::CartridgeData data = romLoader.getData();
-    for (int i = 0; i < 512; i++) {
+    for (uint16_t i = 0x0000; i < 0x0100; i++) {
```
Comment (Sgregory42) [inline test/romloader.t.cpp:51]: sa,e here no magic numbers

**Review by Sgregory42 — CHANGES_REQUESTED** [review]

File: src/romloader.cpp (hunk 5 above)
Comment (MarjorieLV) [reply src/romloader.cpp:19]: Can't find why I did this, putting it on toto

File: test/romloader.t.cpp (hunk 6 above)
Comment (MarjorieLV) [reply test/romloader.t.cpp:51]: can't find why i didi this putting it in TODO

**Pull request merged**